table-api/src/main/java/io/deephaven/api/Strings.java

```diff
@@ -0,0 +1,87 @@
+package io.deephaven.api;
+
+import java.util.Objects;
+
+/**
+ * A set of static helpers to turn strongly-typed api arguments into their {@link String}
+ * counterparts.
+ */
+public class Strings {
+
+    public static String of(ColumnName columnName) {
+        return columnName.name();
+    }
+
+    public static String of(RawString rawString) {
+        return rawString.value();
+    }
+
+    public static String of(ColumnMatch match) {
+        return String.format("%s==%s", of(match.left()), of(match.right()));
+    }
+
+    public static String of(ColumnFormula formula) {
+        return String.format("%s=%s", of(formula.newColumn()), of(formula.expression()));
+    }
+
+    public static String of(ColumnAssignment assignment) {
+        return String.format("%s=%s", of(assignment.newColumn()), of(assignment.existingColumn()));
+    }
+
+    public static String of(Selectable selectable) {
+        return selectable.walk(new UniversalAdapter()).getOut();
+    }
+
+    public static String of(Expression expression) {
+        return expression.walk(new UniversalAdapter()).getOut();
+    }
+
+    public static String of(Filter filter) {
+        return filter.walk(new UniversalAdapter()).getOut();
+    }
+
+    public static String of(JoinAddition joinAddition) {
+        return joinAddition.walk(new UniversalAdapter()).getOut();
+    }
+
+    public static String of(JoinMatch joinMatch) {
+        return joinMatch.walk(new UniversalAdapter()).getOut();
+    }
+
+    /**
+     * If we ever need to provide more specificity for a type, we can create a non-universal impl.
```

Comment:
javadoc should first explain what this class is for, before talking about future ideas...